### PR TITLE
docs: improve documentation about SensitiveEndpointRule replacement

### PR DESCRIPTION
**Pull request by sdelamo**
close: https://github.com/micronaut-projects/micronaut-security/issues/606
close: https://github.com/micronaut-projects/micronaut-security/pull/589

@oscarcitoz this PR refactors `SensitiveEndpointRule` to ease replacement. Moreover, it documents how to restrict management endpoints for a particular role.

### CODE REVIEW

**Review by oscarcitoz — APPROVED** [review]
approved, it hurts that it could not be done with my PR because I did it so that it was not necessary to create a new class to specify the role but from configurations (from applicaiton.yml). anyway thank you very much

File: security/src/main/java/io/micronaut/security/rules/SensitiveEndpointRule.java [hunk 1]
```diff
@@ -57,35 +76,122 @@
...
+                                       @NonNull ExecutableMethod<?, ?> method) {
+        Boolean sensitive = endpointMethods.get(method);
+        if (sensitive) {
+            if (claims == null) {
+                return checkSensitiveAnonymous(request, method);
+            }
+            return checkSensitiveAuthenticated(request, claims, method);
+        }
+        return checkNotSensitive(request, claims, method);
+    }
```
Comment (ilopmar) [inline security/src/main/java/io/micronaut/security/rules/SensitiveEndpointRule.java:113]: I'm not sure if this is the best way to encapsulate everything calling the threes protected methods from here. All three methods are the same in the sense that they log a message and return the appropriate `SecurityRuleResult` but they don't check any condition. Everything is decided in this method.
Did you create those three methods in other for users to be able to override them without override this one?

Comment (sdelamo) [reply security/src/main/java/io/micronaut/security/rules/SensitiveEndpointRule.java:113]: Yes, three protected methods makes easier the replacement. See example, in the tests which I show in the docs.

Comment (ilopmar) [reply security/src/main/java/io/micronaut/security/rules/SensitiveEndpointRule.java:113]: :+1:

**Review by ilopmar — APPROVED** [review]

**Comment by sdelamo** [comment]
thanks @oscarcitoz for pointing us to improving this part of the security module. Really valuable feedback.

**Pull request merged**